### PR TITLE
add/remove style elements on global toggle

### LINKED ISSUE #987: "Turn all styles off" state doesn't work initially in Firefox after toggling "View > Page Style"
* **Browser**: Firefox 78.0.2 (64-bit)
* **Operating System**: Windows 10
* **Stylus Version**: 1.5.11

In Firefox, "Turn all styles off" doesn't work initially for the active tab after toggling `View > Page Style`:
 
1. Check "Turn all styles off" (no styles are applied)
2. Select `View > Page Style > No Style` from the menu bar
3. Select `View > Page Style > Basic Page Style` from the menu bar
4. Observe that styles are applied now for the active tab, even though "Turn all styles off" is checked
5. "Turn all styles off" must be unchecked and then checked again to have an effect for the active tab

### PR DESCRIPTION
**Pull request by tophf**
Fixes #987

### CODE REVIEW

File: content/style-injector.js [hunk 1]
```diff
@@ -19,90 +19,91 @@ self.createStyleInjector = self.INJECTED === 1 ? self.createStyleInjector : ({
...
+      return (
+        !styles.length ?
+          Promise.resolve([]) :
+          docRootObserver.evade(() => {
+            if (!isTransitionPatched && isEnabled) {
+              _applyTransitionPatch(styles);
+            }
+            return styles.map(_addUpdate);
+          })
+      ).then(_emitUpdate);
```
Comment (tophf) [inline content/style-injector.js:38]: Previously, _emitUpdate wasn't called on an empty list which I think was wrong. To simplify the Promise chain _emitUpdate return value is its parameter.

**Comment by narcolepticinsomniac** [comment]
Are we waiting for 8 to review this?

**Comment by tophf** [comment]
Not really because the change is functionally trivial, but I thought it wouldn't hurt.

**Comment by narcolepticinsomniac** [comment]
Alright. Well, he's been around in the meantime, so if he cared, he probably would've reviewed it.

**Comment by narcolepticinsomniac** [comment]
@tophf I might've jumped the gun on this one. I thought I tested thoroughly, but apparently I missed something. Global toggle seems to work fine. I also tested toggling individually, using Github Dark, which also worked fine. Come to find out, other styles toggled individually - not so much.

Not sure if it's an issue with non-usercss or what, but most of my other styles aren't toggling back on once disabled.

2020-07-16 18_16_54-Extensions - Cent Browser

**Comment by tophf** [comment]
Yeah, looking at my new code I see now it makes an incorrect assumption about parent-child relation, I'll open a new PR.

**Pull request merged**